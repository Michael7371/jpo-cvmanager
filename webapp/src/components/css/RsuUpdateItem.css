#updateitem {
  display: flex;
  flex-flow: row nowrap;
  justify-content: center;
}

#ipaddr {
  font-family: Arial, Helvetica, sans-serif;
  color: white;
  margin-top: 10px;
  margin-bottom: 10px;
  margin-left: auto;
  margin-right: auto;
}

#disabledbtn {
  font-family: Arial, Helvetica, sans-serif;
  font-weight: 550;
  background-color: #c0c0c0;
  border: none;
  color: black;
  padding: 10px 10px;
  text-align: center;
  font-size: 12px;
  border-radius: 3px;
  margin-right: 10px;
  margin-left: 15px;
}

#updatebtn {
  font-family: Arial, Helvetica, sans-serif;
  font-weight: 550;
  background-color: #d16d15;
  border: none;
  color: white;
  padding: 10px 10px;
  text-align: center;
  font-size: 12px;
  border-radius: 3px;
  margin-right: 10px;
  margin-left: 15px;
}
