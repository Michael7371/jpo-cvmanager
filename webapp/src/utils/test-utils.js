/**
 * This function searches for the every somewhat random/chaotic class name and property that cause snapshot tests to be inconsistent.
 * This current list includes MUI classes (css-*) and aria-invalid attributes.
 *
 * @ex :
 * ```
 * const { container } = render(<Component />);
 *
 * replaceChaoticIds(container);
 * ```
 *
 * @param container The HTMLElement node to search for SSR ids
 */
function replaceChaoticIds(container) {
  const props = [
    {
      selector: 'class',
      updateFunc: (val) => val.replace(/css-[0-9a-z]*?-/g, 'css-mocked-'),
    },
  ]

  container.querySelectorAll('*').forEach((item) => {
    props.forEach((prop) => {
      if (item.getAttribute(prop.selector)) {
        item.setAttribute(prop.selector, prop.updateFunc(item.getAttribute(prop.selector)))
      }
    })
  })
  container.querySelectorAll('input[aria-invalid]').forEach((item) => {
    item.removeAttribute('aria-invalid')
  })
  return container
}

export { replaceChaoticIds }
