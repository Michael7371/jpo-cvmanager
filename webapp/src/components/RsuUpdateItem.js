import React, { Component } from 'react'
import './css/RsuUpdateItem.css'

class RsuUpdateItem extends Component {
  handleOSClick = () => {
    this.props.handleUpdateOS(this.props.ip)
  }

  handleFWClick = () => {
    this.props.handleUpdateFW(this.props.ip)
  }

  render() {
    return (
      <div id="updateitem">
        <div id="item">
          {<h3 id="ipaddr">{this.props.ip}</h3>}

          {this.props.osUpdateAvailable.includes(this.props.ip) ? (
            <button id="updatebtn" onClick={this.handleOSClick}>
              Update OS
            </button>
          ) : (
            <button id="disabledbtn" disabled={true}>
              Update OS
            </button>
          )}

          {!this.props.osUpdateAvailable.includes(this.props.ip) &&
          this.props.fwUpdateAvailable.includes(this.props.ip) ? (
            <button id="updatebtn" onClick={this.handleFWClick}>
              Update Firmware
            </button>
          ) : (
            <button id="disabledbtn" disabled={true}>
              Update Firmware
            </button>
          )}

          {this.props.osUpdateAvailable.includes(this.props.ip) ||
          this.props.fwUpdateAvailable.includes(this.props.ip) ? (
            <p id="infotext">Allow for 10 minutes for update to occur once started</p>
          ) : null}
        </div>
      </div>
    )
  }
}

export default RsuUpdateItem
