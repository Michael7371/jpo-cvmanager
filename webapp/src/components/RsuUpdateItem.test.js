import React from 'react'
import { render } from '@testing-library/react'
import RsuUpdateItem from './RsuUpdateItem'
import { replaceChaoticIds } from '../utils/test-utils'

it('should take a snapshot', () => {
  const { container } = render(<RsuUpdateItem osUpdateAvailable={[]} fwUpdateAvailable={[]} />)

  expect(replaceChaoticIds(container)).toMatchSnapshot()
})
