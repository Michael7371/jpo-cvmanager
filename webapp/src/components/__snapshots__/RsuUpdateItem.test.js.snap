// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should take a snapshot 1`] = `
<div>
  <div
    id="updateitem"
  >
    <div
      id="item"
    >
      <h3
        id="ipaddr"
      />
      <button
        disabled=""
        id="disabledbtn"
      >
        Update OS
      </button>
      <button
        disabled=""
        id="disabledbtn"
      >
        Update Firmware
      </button>
    </div>
  </div>
</div>
`;
